### PR TITLE
add sample pod spec for NF

### PR DESCRIPTION
**Pull request by ahalimx86**
add sample pod spec for NF and udpate NAD for it to use.

### CODE REVIEW

**Review by paulfhncw — APPROVED** [review]

File: e2e/artefacts/k8s/nad.yaml [hunk 1]
```diff
@@ -1,20 +1,12 @@
 apiVersion: "k8s.cni.cncf.io/v1"
 kind: NetworkAttachmentDefinition
 metadata:
-  name: sriov-net1
+  name: nfnet
   annotations:
     k8s.v1.cni.cncf.io/resourceName: intel.com/ipu
 spec:
   config: '{
   "type": "nf-cni",
```
Comment (palade) [inline e2e/artefacts/k8s/nad.yaml:9]: nf-cni was the name we used in our own implementation of nf-cni. This needs to be the same as the one deployed on in the cluster.

File: e2e/artefacts/k8s/nad.yaml [hunk 2]
```diff
@@ -1,20 +1,12 @@
 apiVersion: "k8s.cni.cncf.io/v1"
 kind: NetworkAttachmentDefinition
 metadata:
-  name: sriov-net1
+  name: nfnet
   annotations:
     k8s.v1.cni.cncf.io/resourceName: intel.com/ipu
```
Comment (palade) [inline e2e/artefacts/k8s/nad.yaml:6]: Is this the resourceName settled by the device plugin? Just to double check

**Review by palade — COMMENTED** [review]
Couple of questions

File: e2e/artefacts/k8s/pod-nf.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  containers:
+  - name: nf 
+    image: intel-networkfunction:latest
+    imagePullPolicy: IfNotPresent
+    securityContext:
+      privileged: true
+    command: [ "/app/nf"]
+    resources:
+      requests:
+        intel.com/ipu: '1'
```
Comment (palade) [inline e2e/artefacts/k8s/pod-nf.yaml:18]: Should be 2 here - as we attach two APFs per NF

**Review by palade — CHANGES_REQUESTED** [review]

File: e2e/artefacts/k8s/nad.yaml (hunk 2 above)
Comment (ahalimx86) [reply e2e/artefacts/k8s/nad.yaml:6]: > Is this the resourceName used by the device plugin? Just to double check
No, this is just name of the NAD.

File: e2e/artefacts/k8s/nad.yaml (hunk 1 above)
Comment (ahalimx86) [reply e2e/artefacts/k8s/nad.yaml:9]: > nf-cni was the name we used in our own implementation of nf-cni. This needs to be the same as the one deployed on in the cluster.

This is just a place-holder for now for whatever cni binary is called for attaching the apf for NF. Once we know from DPU operator we can update it later

File: e2e/artefacts/k8s/pod-nf.yaml (hunk 3 above)
Comment (ahalimx86) [reply e2e/artefacts/k8s/pod-nf.yaml:18]: > Should be 2 here - as we attach two APFs per NF. Also 2 for the limits label

Yes, it should 2. I will update it.

**Review by palade — APPROVED** [review]

**Pull request merged**